### PR TITLE
Dev2

### PR DESCRIPTION
**Pull request by Nastyshka14**

### CODE REVIEW

**Review by Nastyshka14 — COMMENTED** [review]
everything is fixed

File: calendar/README.md [hunk 1]
```diff
@@ -1,70 +1,13 @@
-# Getting Started with Create React App
+# Birthday notifications app
```
Comment (s-severin) [inline calendar/README.md]: Remove this file

File: README.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Birthday notifications app
```
Comment (s-severin) [inline README.md:1]: Fix file description

File: calendar/src/App.css [hunk 3]
```diff
@@ -1,38 +0,0 @@
-.App {
```
Comment (s-severin) [inline calendar/src/App.css:1]: Remove empty file

File: calendar/src/App.tsx [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
-import { BrowserRouter } from 'react-router-dom';
-
+import React from "react";
+import "./App.scss";
+import { CalendarPage } from "./components/Calendar/CalendarPage";
 
 function App() {
   return (
-    <BrowserRouter>
+    <div className="container">
```
Comment (s-severin) [inline calendar/src/App.tsx]: Removce unused class names

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 5]
```diff
@@ -1,80 +1,73 @@
...
+import "./CalendarPage.scss";
+import { Calendar, notification } from "antd";
 import "antd/dist/antd.css";
-import { useContentful } from "../../useContentful.ts";
+import { useContentful } from "../../useContentful";
 import type { Moment } from "moment";
 import moment from "moment";
-import dayjs from "dayjs";
-import { postEntry } from "../../postContentful.js";
+import { useCallback } from "react";
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: Fix imports order

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 6]
```diff
@@ -1,80 +1,73 @@
...
 moment.updateLocale("en", { week: { dow: 1 } });
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: Move to separate config const file
(app/contants/moment.ts)

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 7]
```diff
@@ -1,80 +1,73 @@
...
 
 interface ListDataItem {
-  isRead: boolean;
   name: string;
   date: string;
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: https://stackoverflow.com/questions/45485073/how-do-i-express-a-date-type-in-typescript

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 8]
```diff
@@ -1,80 +1,73 @@
...
-import { useContentful } from "../../useContentful.ts";
+import { useContentful } from "../../useContentful";
 import type { Moment } from "moment";
 import moment from "moment";
-import dayjs from "dayjs";
-import { postEntry } from "../../postContentful.js";
+import { useCallback } from "react";
 moment.updateLocale("en", { week: { dow: 1 } });
 
 interface ListDataItem {
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: `interface ListDataItem = Array<BirthdayItem>`

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 9]
```diff
@@ -1,80 +1,73 @@
...
-
-  const handleOk = () => {
-    setIsModalVisible(false);
-  };
 
-  const handleCancel = () => {
-    setIsModalVisible(false);
-  };
+  const openNotification = useCallback((item: string) => {
+    const args = {
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: Add TS description for const

Comment (s-severin) [reply calendar/src/components/Calendar/CalendarPage.tsx]: `const args: CalendarNotification = {`

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 10]
```diff
@@ -1,80 +1,73 @@
...
+  }, []);
 
   useEffect(() => {
-    getBirthdays().then((response) => setBirthdays(response));
-  }, []);
+    getBirthdays().then((data) => setBirthdays(data));
+  }, [getBirthdays]);
 
   const getListData = (value: Moment) => {
     const listData: ListDataItem[] = [];
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: const listData: ListDataItem

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 11]
```diff
@@ -1,80 +1,73 @@
...
-
-    birthays.forEach((element) => {
-      if (element.date === value.format("YYYY-MM-DD")) {
+    birthdays.forEach((element: BirthdayItem) => {
+      if (
+        element.date === moment(value.format("YYYY-MM-DD")).toISOString(true)
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: move to const `moment(value.format("YYYY-MM-DD")).toISOString(true)`

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 12]
```diff
@@ -1,80 +1,73 @@
...
   };
 
-  const getReminder = () => {
-    birthays.forEach((item) => {
-      if (item.date === dayjs().format("YYYY-MM-DD")) {
-        showModal()
+  const getReminder = useCallback(() => {
+    birthdays.forEach((item: BirthdayItem) => {
+      if (
+        item.date === moment(moment().format("YYYY-MM-DD")).toISOString(true)
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: To other one const `moment(moment().format("YYYY-MM-DD")).toISOString(true)`

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 13]
```diff
@@ -1,80 +1,73 @@
...
     });
     return listData || [];
   };
 
-  const getReminder = () => {
-    birthays.forEach((item) => {
-      if (item.date === dayjs().format("YYYY-MM-DD")) {
-        showModal()
+  const getReminder = useCallback(() => {
+    birthdays.forEach((item: BirthdayItem) => {
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: birthdays -> listData

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 14]
```diff
@@ -1,80 +1,73 @@
...
+  }, [getReminder]);
 
-  const dateCellRender = (value: any) => {
-    let listData = getListData(value);
+  const dateCellRender = (value: Moment) => {
+    let listData: ListDataItem[] = getListData(value);
     return (
       <ul className="events">
-        {listData.map((item: any) => (
+        {listData.map((item: ListDataItem) => (
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: listData check length

File: calendar/src/components/Calendar/CalendarPage.tsx [hunk 15]
```diff
@@ -84,14 +77,8 @@ export const CalendarPage = () => {
...
-    <>
-      <Calendar dateCellRender={dateCellRender} />
-      <Modal
-        title="Basic Modal"
-        visible={isModalVisible}
-        onOk={handleOk}
-        onCancel={handleCancel}
-      ></Modal>
-    </>
+    <div className="calendar">
```
Comment (s-severin) [inline calendar/src/components/Calendar/CalendarPage.tsx]: Add styles

File: calendar/src/setupTests.js [hunk 16]
```diff
@@ -1,5 +1 @@
-// jest-dom adds custom jest matchers for asserting on DOM nodes.
-// allows you to do things like:
-// expect(element).toHaveTextContent(/react/i)
-// learn more: https://github.com/testing-library/jest-dom
 import '@testing-library/jest-dom';
```
Comment (s-severin) [inline calendar/src/setupTests.js]: Add some test

File: calendar/src/useContentful.tsx [hunk 17]
```diff
@@ -0,0 +1,44 @@
+interface BirthdayItem {
+  name: string;
+  date: string;
```
Comment (s-severin) [inline calendar/src/useContentful.tsx]: Date

File: calendar/src/useContentful.tsx [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+const query = `
+{
+  birthdaysCollection {
```
Comment (s-severin) [inline calendar/src/useContentful.tsx]: image

File: calendar/src/useContentful.tsx [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+export const useContentful = () => {
+  const getBirthdays = async (): Promise<BirthdayItem[]> => {
+    try {
+      return fetch(
+        `https://graphql.contentful.com/content/v1/spaces/g5vcvjt0rgq7/`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: "Bearer L482r6IOCJjaooW7x2WDVsyi4XOVy5cHWlFmINREF9w",
```
Comment (s-severin) [inline calendar/src/useContentful.tsx]: TOKEN TO .ENV

File: calendar/src/useContentful.tsx [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+          },
+          body: JSON.stringify({ query }),
+        }
+      )
+        .then((response) => response.json())
+        .then(({ data, error }) => {
+          if (error) {
+            throw new Error("Error fetching birthdays");
+          }
+          return data.birthdaysCollection.items;
```
Comment (s-severin) [inline calendar/src/useContentful.tsx]: Check if not have birthdaysCollection or items

File: calendar/tsconfig.json [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "allowSyntheticDefaultImports": true,
+    "target": "es6",
+    "sourceMap": true,
+    "outDir": "./dist",
+    "baseUrl": "./",
+    "incremental": true,
+    "jsx": "preserve",
+    "allowJs": true
```
Comment (s-severin) [inline calendar/tsconfig.json]: "allowJs": false

**Review by s-severin — CHANGES_REQUESTED** [review]

File: calendar/README.md (hunk 1 above)
Comment (Nastyshka14) [reply calendar/README.md]: Done

File: README.md (hunk 2 above)
Comment (Nastyshka14) [reply README.md:1]: Done

File: calendar/src/App.css (hunk 3 above)
Comment (Nastyshka14) [reply calendar/src/App.css:1]: Done

File: calendar/src/App.tsx (hunk 4 above)
Comment (Nastyshka14) [reply calendar/src/App.tsx]: Done

File: calendar/src/components/Calendar/CalendarPage.tsx (hunk 5 above)
Comment (Nastyshka14) [reply calendar/src/components/Calendar/CalendarPage.tsx]: Done rearranged
But forgot what to do with moment.js😏

File: calendar/src/useContentful.tsx (hunk 17 above)
Comment (Nastyshka14) [reply calendar/src/useContentful.tsx]: Done

File: calendar/src/useContentful.tsx (hunk 18 above)
Comment (Nastyshka14) [reply calendar/src/useContentful.tsx]: Moved to a constant
But I could not rename it

File: calendar/src/useContentful.tsx (hunk 19 above)
Comment (Nastyshka14) [reply calendar/src/useContentful.tsx]: Done

File: calendar/src/useContentful.tsx (hunk 20 above)
Comment (Nastyshka14) [reply calendar/src/useContentful.tsx]: Done

File: calendar/tsconfig.json (hunk 21 above)
Comment (Nastyshka14) [reply calendar/tsconfig.json]: Done

**Pull request closed without merging**